src/app/screens/confirmNftTransaction/index.tsx

```diff
@@ -150,8 +152,27 @@ function ConfirmNftTransaction() {
 
   const handleOnConfirmClick = (txs: StacksTransaction[]) => {
     if (isLedgerAccount(selectedAccount)) {
-      const txType: LedgerTransactionType = 'STX';
-      navigate('/confirm-ledger-tx', { state: { unsignedTx, type: txType } });
+      const type: LedgerTransactionType = 'STX';
+      navigate('/confirm-ledger-tx', {
+        state: {
+          unsignedTx: txs[0].serialize(),
+          type,
+          recipients: [
+            {
+              address: recipientAddress,
+              amountSats: initialStxTransactions[0]?.payload?.amount
+                ? new BigNumber(initialStxTransactions[0]?.payload.amount?.toString(10))
+                : new BigNumber(0),
+            },
+          ],
+          fee: new BigNumber(
+            initialStxTransactions
```

Comment:
how come we use initialStxTransactions[0] here instead of txs[0]?

Reply:
oh because you pass unsigned txs to ledger confirm screen...

Reply:
is there an assumption that ledger does not support multisig, as the soft wallets do?

Reply:
Nice catch! It looks like I overlooked it and propagated it to several places, it should be an initial unsigned tx

Reply:
nice, saw the fixes. I think just need to address the double signing part, if I am reading the code correctly:

> there is some ledger signing logic already in confirmStxTransactionComponent, but the confirmLedgerTransaction component signs it again. is this double signing expected?

Reply:
The `src/app/components/confirmStxTransactionComponent/index.tsx` component is being reused in multiple places, and as it's used for both regular transactions (our extension full-screen tab) and incoming transactions (our popup on external websites), we have different flow for signing for these 2 cases. and you can see the skipModal prop there which is used to skip the modal signing for the full-screen tab, that's why there's no double-signing

Reply:
ok makes sense